Framework/Core/include/Framework/RawDeviceService.h

```diff
@@ -0,0 +1,33 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef FRAMEWORK_RAWDEVICESERVICE_H
+#define FRAMEWORK_RAWDEVICESERVICE_H
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace o2 {
+namespace framework {
+
+/// This service provides a hook into the actual fairmq device running the
+/// computation, and allows and advanced user to modify its behavior in
+/// from with a workflow class. This should be used to implement special
+/// `DataProcessors` like one that acts as a gateway to standard FairMQ
+/// devices.
+class RawDeviceService {
```

Comment:
According to the coding guidelines, this purely abstract class should be suffixed with _Interface_ (https://htmlpreview.github.io/?https://github.com/AliceO2Group/CodingGuidelines/blob/master/naming_formatting.html#Interface_Names)

That might apply to other of your services, I did not check.

Reply:
A "purely abstract class" do not exist as a concept in C++. The only think that exists is an "abstract class" (i.e. one which has at least one virtual method). "Pure Interface" or "purely abstract class" is Java speak, so I personally find our Coding Guidelines misleading.

That said, in this particular case, I do not think I would call this an Interface, as the fact that the base implementation only has pure virtual methods is merely an accident, not a design decision. I might at some point implement some non pure virtual method or have some one of the services use a base class which does not have only pure methods (for example if I expose Monitoring as a Service, as I was planning to do). This would lead to an inconsistent API were some of the services have to be retrieved via:

```cpp
services().get<SomeServiceInterface>()
```

while others:

```cpp
services().get<SomeOtherService>()
```

I actually now checked the original [C++ Coding Guidelines](https://google.github.io/styleguide/cppguide.html#Interfaces) from Google and indeed they define the rule of a set of requirements for *optionally* adopt the `Interface` suffix, not as a mandatory naming convention. To me that's the sensible, C++ "native", rule to adopt and I would propose we actually change the Guidelines to match the original spirit.

Reply:
I've created a PR in AliceO2Group/CodingGuidelines#4. We can move the discussion there, if needed.

Reply:
Agreed with the intent here and with the amended coding guidelines